src/Worker.Extensions.DurableTask/DurableTaskFunctionsMiddleware.cs

```diff
@@ -120,7 +128,24 @@ public FunctionsOrchestrationContext(TaskOrchestrationContext innerContext, Func
         public override T GetInput<T>()
         {
             this.EnsureLegalAccess();
-            return this.innerContext.GetInput<T>()!;
+
+            // TODO: address this hack.
+            // The default TaskOrchestrationContext is not actually dynamic with GetInput - it was set
+            // once based on the declared input type of the orchestrator. Since we do not know the
+            // desired input type upfront, we were initialized to object. So we must serialize and
+            // deserialize again to convert to our desired type.
+            if (this.serializedInput is null)
+            {
+                object? input = this.innerContext.GetInput<object>();
+                if (input is null)
+                {
+                    return default!;
+                }
+
+                this.serializedInput = this.options.DataConverter.Serialize(input);
```

Comment:
Is there a less hacky way to do this, like converting the deserialized `JsonElement` into the target type?

Reply:
Refactored this to include an optimal path for `JsonElement`. I retained the serialization logic because we are not guaranteed it will be `JsonElement`, the result of the `object` deserialization depends on what serializer they have configured.